docs/design/2022-05-17-new-implementation-of-mpp-partition-table-support.md

```diff
@@ -0,0 +1,130 @@
+| └─PartitionUnion_14                | 2.00     | root      |                               |                                   |
+|   ├─StreamAgg_29                   | 1.00     | root      |                               | funcs:count(Column#12)->Column#10 |
+|   │ └─TableReader_30               | 1.00     | root      |                               | data:StreamAgg_18                 |
+|   │   └─StreamAgg_18               | 1.00     | cop[tikv] |                               | funcs:count(1)->Column#12         |
+|   │     └─Selection_28             | 3333.33  | cop[tikv] |                               | gt(test.employees.store_id, 10)   |
+|   │       └─TableFullScan_27       | 10000.00 | cop[tikv] | table:employees, partition:p2 | keep order:false, stats:pseudo    |
+|   └─StreamAgg_48                   | 1.00     | root      |                               | funcs:count(Column#14)->Column#10 |
+|     └─TableReader_49               | 1.00     | root      |                               | data:StreamAgg_37                 |
+|       └─StreamAgg_37               | 1.00     | cop[tikv] |                               | funcs:count(1)->Column#14         |
+|         └─Selection_47             | 3333.33  | cop[tikv] |                               | gt(test.employees.store_id, 10)   |
+|           └─TableFullScan_46       | 10000.00 | cop[tikv] | table:employees, partition:p3 | keep order:false, stats:pseudo    |
++------------------------------------+----------+-----------+-------------------------------+-----------------------------------+
+
+```
+
+### Dynamic partition prune
+
+In this mode, partition prune happens when TiDB generates executor. Compared to static partition prune, dynamic partition prune has at least two advantages:
+
+- The plan is more elegant
+- Using dynamic partition prune, TiDB can use some runtime information to do the partition prune. For example, in index join, if the probe side is a partition table, TiDB can use the data from build side to prune the partition.
+
+A typical plan under dynamic partition prune is
+
+```sql
+mysql> explain select count(*) from employees where store_id > 10;
++------------------------------+----------+-----------+-----------------+----------------------------------+
+| id                           | estRows  | task      | access object   | operator info                    |
++------------------------------+----------+-----------+-----------------+----------------------------------+
+| StreamAgg_20                 | 1.00     | root      |                 | funcs:count(Column#11)->Column#9 |
+| └─TableReader_21             | 1.00     | root      | partition:p2,p3 | data:StreamAgg_9                 |
+|   └─StreamAgg_9              | 1.00     | cop[tikv] |                 | funcs:count(1)->Column#11        |
+|     └─Selection_19           | 3333.33  | cop[tikv] |                 | gt(test.employees.store_id, 10)  |
```

Comment:
it seems the engine is cop（tikv），is it expected？

Reply:
Yes, it is in the `background` session.